### PR TITLE
make warnings more verbose

### PR DESCRIPTION
**Pull request by carlculator**

### CODE REVIEW

File: timeseriesx/mixins/time_zone.py [hunk 1]
```diff
@@ -7,6 +7,10 @@
 )
 
 
+class TimeZoneWarning(Warning):
```
Comment (TimoGuenther) [inline timeseriesx/mixins/time_zone.py]: `TimeZoneWarning` and `UnitWarning` could be a subclasses of `RuntimeWarning`.

File: timeseriesx/mixins/time_zone.py [hunk 2]
```diff
@@ -7,6 +7,10 @@
...
+    pass
```
Comment (TimoGuenther) [inline timeseriesx/mixins/time_zone.py:14]: A short docstring explaining the intended use and purpose of `TimeZoneWarning` and `UnitWarning` would be helpful.

File: timeseriesx/base/timestamp_series.py [hunk 3]
```diff
@@ -410,7 +410,8 @@ def _basic_calc_time_series(self, operation, other, **kwargs):
         if not self.unit == other.unit:
             raise ValueError("The time series have different units")
         if not self._series.index.equals(other._series.index):
-            warnings.warn("timestamps do not match, values are auto-filled")
+            warnings.warn("timestamps do not match, values are auto-filled",
+                          category=RuntimeWarning)
```
Comment (TimoGuenther) [inline timeseriesx/base/timestamp_series.py]: For improved traceability, the category could be a custom warning class like `TimeZoneWarning` and `UnitWarning`.

**Review by TimoGuenther — CHANGES_REQUESTED** [review]

File: timeseriesx/base/timestamp_series.py (hunk 3 above)
Comment (carlculator) [reply timeseriesx/base/timestamp_series.py]: I don't have a good name for it. Any suggestions?

Comment (TimoGuenther) [reply timeseriesx/base/timestamp_series.py]: Perhaps `TimestampMismatchWarning` or `MismatchedTimestampsWarning`.

File: timeseriesx/mixins/time_zone.py (hunk 1 above)
Comment (carlculator) [reply timeseriesx/mixins/time_zone.py]: ok. done.

File: timeseriesx/mixins/time_zone.py (hunk 2 above)
Comment (carlculator) [reply timeseriesx/mixins/time_zone.py:14]: ok. done.

File: timeseriesx/base/timestamp_series.py (hunk 3 above)
Comment (carlculator) [reply timeseriesx/base/timestamp_series.py]: thanks. applied it

**Comment by TimoGuenther** [comment]
This merge was premature considering the changes were not endorsed yet. Indeed, I would propose further changes: The `pass` below the docstrings are redundant because `pass` is only syntactical.

**Pull request merged**